xonsh/aliases.py

```diff
@@ -313,6 +314,37 @@ def bang_bang(args, stdin=None):
     return bang_n(['-1'])
 
 
+def which(args, stdin=None):
+    """
+    Checks if argument is a xonsh alias, then if it's an executable, then
+    finally throw an error.
+    If '-a' flag is passed, run both to return both `xonsh` match and
+    `which` match
+    """
+    match = set(args).intersection(builtins.aliases)
+    #skip alias check if user asks a skip or the version number
+    if (match and
+      not '--skip-alias' in args and
+      not {'--version','-v','-V'}.intersection(set(args))):
```

Comment:
Is there any reason not to use argparse? This isn't blocking, I am just curious...